language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ConditionalExpressionNodeContext.java

```diff
@@ -44,12 +52,50 @@ public ConditionalExpressionNodeContext() {
     public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context, ConditionalExpressionNode node)
             throws LSCompletionException {
         NonTerminalNode nodeAtCursor = context.getNodeAtCursor();
-        if (this.onQualifiedNameIdentifier(context, nodeAtCursor)) {
+        int cursor = context.getCursorPositionInTree();
+        int colonTokenPos = node.colonToken().textRange().startOffset();
+        List<LSCompletionItem> completionItems = new ArrayList<>();
+
+        if (cursor > colonTokenPos && node.middleExpression().kind() == SyntaxKind.SIMPLE_NAME_REFERENCE 
+                && node.endExpression().kind() == SyntaxKind.SIMPLE_NAME_REFERENCE) {
+            /*
+            Covers the following context
+            eg: int n = true ? module1:
+             */
+            String middleExprName = ((SimpleNameReferenceNode) node.middleExpression()).name().text();
+            String alias = middleExprName.startsWith("'") ? middleExprName.substring(1) : middleExprName;
```

Comment:
Isn't the `'` handled by default? Do we need to check it here specifically? I think the module name consists the name with the single quote

Reply:
there is a new overload method `getExpressionContextEntries` introduced in `QNameReferenceUtil.java` which does not take a `QualifiedNameReferenceNode` but just the `String moduleAlias`. So it is not handled by default.

Reply:
Then, shall we move this check to the new overloaded method? So that would be common. WDYT?

Reply:
`'` is handled in `QNameReferenceUtil.getAlias` which takes an `QualifiedNameReferenceNode`, but in this case, we do not have a `QualifiedNameReferenceNode` unless we create it using `NodeFactory`

Reply:
Changes added with the commit 8048749